js/components/staking-modal-new.js

```diff
@@ -674,7 +674,20 @@ class StakingModalNew {
             // Show error notification
             if (window.notificationManager) {
                 const errorMsg = error.message || 'Failed to approve tokens';
-                window.notificationManager.show(`Approval failed: ${errorMsg}`, 'error');
+                let userFriendlyMessage = 'Failed to approve tokens';
+                
+                // Extract user-friendly message from error
+                if (errorMsg.includes('user rejected') || errorMsg.includes('user denied')) {
+                    userFriendlyMessage = 'Transaction cancelled by user';
+                } else if (errorMsg.includes('insufficient funds')) {
+                    userFriendlyMessage = 'Insufficient funds for transaction';
+                } else if (errorMsg.includes('network')) {
+                    userFriendlyMessage = 'Network connection issue';
+                } else if (errorMsg.includes('gas')) {
+                    userFriendlyMessage = 'Gas estimation failed';
+                }
+                
+                window.notificationManager.error(`Approval failed: ${userFriendlyMessage}`);
```

Comment:
probably not for this PR, but looks like this approve isn't using the contract manager so it is not going through the error handler

Reply:
added to todo list